homeassistant/components/modbus/validators.py

```diff
@@ -349,3 +351,29 @@ def duplicate_fan_mode_validator(config: dict[str, Any]) -> dict:
     for key in reversed(errors):
         del config[CONF_FAN_MODE_VALUES][key]
     return config
+
+
+def check_hvac_target_temp_registers(config: dict) -> dict:
+    """Check conflicts among HVAC target temperature registers and HVAC ON/OFF, HVAC register, Fan Modes."""
+    if CONF_HVAC_MODE_REGISTER in config:
+        if CONF_HVAC_MODE_TT_REG_BY_VALUES in config[CONF_HVAC_MODE_REGISTER]:
+            errors = []
+            addresses: list[int] = []
+            name = config[CONF_NAME]
+            addresses.append(int(config[CONF_HVAC_MODE_REGISTER][CONF_ADDRESS]))
+            if CONF_HVAC_ONOFF_REGISTER in config:
+                addresses.append(int(config[CONF_HVAC_ONOFF_REGISTER]))
+            if CONF_FAN_MODE_REGISTER in config:
+                addresses.append(int(config[CONF_FAN_MODE_REGISTER][CONF_ADDRESS]))
+            for key, value in config[CONF_HVAC_MODE_REGISTER][
+                CONF_HVAC_MODE_TT_REG_BY_VALUES
+            ].items():
+                if value in addresses:
+                    wrn = f"In {name} Register {value} already as {CONF_HVAC_ONOFF_REGISTER} or {CONF_HVAC_MODE_REGISTER} or {CONF_FAN_MODE_REGISTER}. {key} is not loaded! "
+                    _LOGGER.warning(wrn)
+                    errors.append(key)
+            for key in reversed(errors):
+                del config[CONF_HVAC_MODE_REGISTER][CONF_HVAC_MODE_TT_REG_BY_VALUES][
+                    key
```

Comment:
For each register, you need to secure it is not used in any other configuration part (as done e.g. with sensors).

Reply:
This specific routine avoids registers overlapping inside the climate. The testing routine is called 3 times with a different register overlapping, so I have tested hvac mode register, hvac on/off and hvac fan mode. 
The only register is admitted to overlap is the target_temp_register, the default one.